test/permission.validation.js

```diff
@@ -37,20 +37,19 @@ metatests.test('Permission validation test', async test => {
       );
     }
     console.error(err);
-    test.end();
-    return;
+    return null;
   }
 
   test.beforeEach(prepareRemoteProvider(ctx.schema));
-  test.afterEach(async (test, callback) => {
-    await test.context.remoteProvider.close();
-    callback();
-  });
+  test.afterEach(async test => test.context.remoteProvider.close());
 
   runTests(test, ctx);
 
-  test.on('done', async () => {
-    await ctx.provider.close();
-    ctx.server.close();
-  });
+  return new Promise(resolve =>
```

Comment:
Why did you wrap this into Promise?
You weren't calling `test.end()` inside the `'done'` event listener before, which would be similar to returning the promise that resolves when this event is emitted. Was it not working without that?

Reply:
@belochub there is a problem with `test.cbFail()` without this promise, as it won't be called. This solution was found through trial and error, so there can be another way to handle it.